plasmapy/particles/particle_class.py

```diff
@@ -2172,6 +2192,12 @@ def charge(self, q: Optional[Union[u.Quantity, Real]]):
             warnings.warn(
                 f"CustomParticle charge set to {q} times the elementary charge."
             )
+            warnings.warn(
+                "The ability to provide a real number to the 'charge' "
+                "parameter to has been deprecated and will be removed "
+                "in a future release of PlasmaPy. Use 'Z' instead.",
+                PlasmaPyDeprecationWarning,
```

Comment:
[Optional] could also do a similar note as before here. But it's arguably less important as the particle creation is the most important point of contact for people for this class.